src/StreamSocket.js

```diff
@@ -0,0 +1,62 @@
+const Transform = require('stream').Transform;
+var d1 = require('diya-sdk');
+
+class StreamSocket extends Transform {
```

Comment:
I would just call this ```Socket```, a socket IS a full-duplex stream so no need to add that info in the name.